test/e2e/storage/vsphere_scale.go

```diff
@@ -0,0 +1,144 @@
+		scDefaultSpec := getVSphereStorageClassSpec("sc-default", nil)
+		scDefault, err := client.StorageV1().StorageClasses().Create(scDefaultSpec)
+		Expect(err).NotTo(HaveOccurred())
+		defer client.StorageV1().StorageClasses().Delete(scDefault.Name, nil)
+		scArrays[0] = scDefault
+
+		// Create Storage Class with vsan storage capabilities
+		By("Creating Storage Class : sc-vsan")
+		var scvsanParameters map[string]string
+		scvsanParameters = make(map[string]string)
+		scvsanParameters[Policy_HostFailuresToTolerate] = "1"
+		scvsanSpec := getVSphereStorageClassSpec("sc-vsan", scvsanParameters)
+		scvsan, err := client.StorageV1().StorageClasses().Create(scvsanSpec)
+		Expect(err).NotTo(HaveOccurred())
+		defer client.StorageV1().StorageClasses().Delete(scvsan.Name, nil)
+		scArrays[1] = scvsan
+
+		// Create Storage Class with SPBM Policy
+		By("Creating Storage Class : sc-spbm")
+		var scSpbmPolicyParameters map[string]string
+		scSpbmPolicyParameters = make(map[string]string)
+		goldPolicy := os.Getenv("VSPHERE_SPBM_GOLD_POLICY")
+		Expect(goldPolicy).NotTo(BeEmpty())
+		scSpbmPolicyParameters[SpbmStoragePolicy] = goldPolicy
+		scSpbmPolicySpec := getVSphereStorageClassSpec("sc-spbm", scSpbmPolicyParameters)
+		scSpbmPolicy, err := client.StorageV1().StorageClasses().Create(scSpbmPolicySpec)
+		Expect(err).NotTo(HaveOccurred())
+		defer client.StorageV1().StorageClasses().Delete(scSpbmPolicy.Name, nil)
+		scArrays[2] = scSpbmPolicy
+
+		// Create Storage Class with User Specified Datastore.
+		By("Creating Storage Class : sc-user-specified-datastore")
+		var scWithDatastoreParameters map[string]string
+		scWithDatastoreParameters = make(map[string]string)
+		datastore := os.Getenv("VSPHERE_DATASTORE")
+		Expect(goldPolicy).NotTo(BeEmpty())
+		scWithDatastoreParameters[Datastore] = datastore
+		scWithDatastoreSpec := getVSphereStorageClassSpec("sc-user-specified-ds", scWithDatastoreParameters)
+		scWithDatastore, err := client.StorageV1().StorageClasses().Create(scWithDatastoreSpec)
+		Expect(err).NotTo(HaveOccurred())
+		defer client.StorageV1().StorageClasses().Delete(scWithDatastore.Name, nil)
+		scArrays[3] = scWithDatastore
+
+		var wg sync.WaitGroup
+		wg.Add(numberOfInstances)
+		volumeCountPerInstance := volumeCount / numberOfInstances
+		for instanceCount := 0; instanceCount < numberOfInstances; instanceCount++ {
+			if instanceCount == numberOfInstances-1 {
+				volumeCountPerInstance = volumeCount
+			}
+			volumeCount = volumeCount - volumeCountPerInstance
+			go CreateVolumesInParallel(client, namespace, scArrays, volumeCountPerInstance, volumesPerPod, &wg)
+		}
+		wg.Wait()
+		//TODO in next task
+		//1. Delete all pods.
+		//2. Delete all PVC's.
+	})
+})
+
+// PerformVolumeLifeCycleAtScale peforms full volume life cycle management at scale
+func PerformVolumeLifeCycleAtScale(client clientset.Interface, namespace string, sc []*storageV1.StorageClass, volumeCountPerInstance int, volumesPerPod int, wg *sync.WaitGroup) {
```

Comment:
Who is using this function?

Reply:
Please see my comment above.

Reply:
Balu .. Lifecycle word means create, attach, detach, delete all ..
I think our objective is just to create the volumes and attach to Pods.
So I think function name should be changed to VolumeCreateAndAttach.

Reply:
Makes sense. I have modified the name.